File: DataAccessLayer/Models/CompanyActivity.cs
using System;
using System.Collections.Generic;

namespace DataAccessLayer.Models
{
    public partial class CompanyActivity
    {
        public virtual int Id { get; set; }
        public virtual int? ActivityId { get; set; }
        public virtual string FirstName { get; set; }
        public virtual string LastName { get; set; }
        public virtual string Email { get; set; }

        public virtual Activity Activity { get; set; }
    }
}

File: OCASWeb/src/app/companyActivity.ts
import { Observable } from 'rxjs/Observable';
import { HttpErrorResponse } from '@angular/common/http';


export class OcasActivitySignup {
  constructor(
    public id: string = '',
    public firstName: string = '',
    public lastName: string = '',
    public email: string = '',
    public activityId: string = '',
    public activityName: string = '',
    public activity: OcasActivity = new OcasActivity()) {
  }

  
}

export class OcasActivity {
  id: number;
  name: string;
}

export interface CallBackAfterError {

  <T>(): Observable<T>;
}


export interface ErrorDisplayHelper {
  <T>(error: HttpErrorResponse, caught: T)
}




